dm/syncer/handle_error.go

```diff
@@ -53,19 +53,22 @@ func (s *Syncer) HandleError(ctx context.Context, req *pb.HandleWorkerErrorReque
 
 	var err error
 	// remove outdated operators when add operator
-	err = s.errOperatorHolder.RemoveOutdated(s.checkpoint.FlushedGlobalPoint())
+	s.streamerController.RemoveOutdated(s.checkpoint.FlushedGlobalPoint().Position)
```

Comment:
Can we remove them when flushing the global checkpoint? Otherwise, will we compare the event position with the operators' every time we get event from it?

Reply:
> will we compare the event position with the operators' every time we get event from it?

we only compare the frontOp which is O(1). And RemoveOutdated is O(lg N) so I hope it can be called in lower frequency as the old logic.

Reply:
lgtm